src/chatd.cpp

```diff
@@ -506,21 +506,30 @@ Promise<void> Connection::reconnect()
                    return;
                 }
 
-                // update DNS cache
-                bool match = mDNScache.set(mUrl.host, ipv4, ipv6);
-
                 if (!cachedIPs) // connect required DNS lookup
                 {
                     CHATDS_LOG_DEBUG("Hostname resolved by first time. Connecting...");
+
+                    mDNScache.set(mUrl.host, ipsv4.at(0), ipsv6.at(0));
                     doConnect();
                     return;
                 }
+
+                string ipv4, ipv6;
+                mDNScache.get(mUrl.host, ipv4, ipv6);
+                bool match = ((std::find(ipsv4.begin(), ipsv4.end(), ipv4) != ipsv4.end())
+                        && (std::find(ipsv6.begin(), ipsv6.end(), ipv6) != ipsv6.end()));
+
                 if (match)
                 {
                     CHATDS_LOG_DEBUG("DNS resolve matches cached IPs.");
                 }
                 else
                 {
+                    // update DNS cache
+                    bool ret = mDNScache.set(mUrl.host, ipv4, ipv6);
```

Comment:
In this line are you setting the same IP's that you got in `mDNScache.get(mUrl.host, ipv4, ipv6);` ?

Reply:
You should set ipsv4.at(0), isn't it?